### PR TITLE
Feature/EUI 8786

### PR DESCRIPTION
**Pull request by DavidJayakumar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-8786

### Change description ###

PostCode validation for UK addresses

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnbenjamin-hmcts — APPROVED** [review]

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

File: src/custom-validators/postcode.validator.spec.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  it('validation should return true, when the first part of postcode is invalid', () => {
+    control.setValue('l15');
+    const validatorFn = postcodeValidator();
+    const validationError = validatorFn(control);
+    expect(validationError).not.toBeNull();
+    expect(validationError.invalidPostcode).toBeTruthy();
+  });
+
+  it('validation should return true, when the secod part of postcode is invalid', () => {
```
Comment (Daniel-Lam) [inline src/custom-validators/postcode.validator.spec.ts:19]: @DavidJayakumar - typo `secod`

File: src/custom-validators/postcode.validator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    control.setValue('l15 555');
+    const validatorFn = postcodeValidator();
+    const validationError = validatorFn(control);
+    expect(validationError).not.toBeNull();
+    expect(validationError.invalidPostcode).toBeTruthy();
+  });
+
+  it('validation should be undefined when valid postcode is enterined lowercase', () => {
```
Comment (Daniel-Lam) [inline src/custom-validators/postcode.validator.spec.ts:27]: @DavidJayakumar - typo `enterined`

File: src/custom-validators/postcode.validator.spec.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    control.setValue('l155ax');
+    const validatorFn = postcodeValidator();
+    const validationError = validatorFn(control);
+    expect(validationError).toBeUndefined();
+  });
+
+  it('validation should be undefined when valid postcode is enterined uppercase with space', () => {
```
Comment (Daniel-Lam) [inline src/custom-validators/postcode.validator.spec.ts:34]: @DavidJayakumar - typo `enterined`

File: src/custom-validators/postcode.validator.spec.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    control.setValue('L15 5AX');
+    const validatorFn = postcodeValidator();
+    const validationError = validatorFn(control);
+    expect(validationError).toBeUndefined();
+  });
+
+  it('validation should be undefined when a valid postcode that is not in postcde database is entered', () => {
```
Comment (Daniel-Lam) [inline src/custom-validators/postcode.validator.spec.ts:41]: @DavidJayakumar - typo `postcde`

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@DavidJayakumar - some minor typos but everything else is OK

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**